src/wrek_event.erl

```diff
@@ -10,6 +12,20 @@
          vert_start/5,
          vert_msg/3]).
 
+
+-spec time_diff(wrek_event(), wrek_event()) -> non_neg_integer().
+
+time_diff(E1, E2) ->
+    time_diff(E1, E2, microsecond).
+
+
+-spec time_diff(wrek_event(), wrek_event(), erlang:time_unit()) ->
+    non_neg_integer().
+
+time_diff(#wrek_event{timestamp = E1}, #wrek_event{timestamp = E2}, TimeUnit) ->
```

Comment:
It might be better to not `E1` and `E2` for both the `timestamp` and `wrek_event`.